Can't build on windows

The rust part of the example and native-cpp were successfully compiled and both working fine

npm run build-cpp-ffi does'not. It can't resolve external symbol "int __cdecl fibonacci(int)"
```
npm run build-cpp-ffi

> embed@1.0.0 build-cpp-ffi C:\local_projects\rust\rust-in-node
> (cd src/native-cpp-ffi && node-gyp configure build)


C:\local_projects\rust\rust-in-node\src\native-cpp-ffi>if not defined npm_config_node_gyp (node "C:\Program Files (x86)\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" configure build )  else (node  configure build )
Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
     Creating library C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\build\Release\addon.lib and object C:\local_projects\
  rust\rust-in-node\src\native-cpp-ffi\build\Release\addon.exp
addon.obj : error LNK2001: unresolved external symbol "int __cdecl fibonacci(int)" (?fibonacci@@YAHH@Z) [C:\local_projects\rust\ru
st-in-node\src\native-cpp-ffi\build\addon.vcxproj]
C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\build\Release\addon.node : fatal error LNK1120: 1 unresolved externals [C:\
local_projects\rust\rust-in-node\src\native-cpp-ffi\build\addon.vcxproj]
gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\MSBuild\14.0\bin\msbuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Program Files (x86)\nodejs\node_modules\npm\node_modules\node-gyp\lib\build.js:270:23)
gyp ERR! stack     at emitTwo (events.js:100:13)
gyp ERR! stack     at ChildProcess.emit (events.js:185:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Windows_NT 10.0.10586
gyp ERR! command "C:\\Program Files (x86)\\nodejs\\node.exe" "C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "configure" "build"
gyp ERR! cwd C:\local_projects\rust\rust-in-node\src\native-cpp-ffi
gyp ERR! node -v v5.5.0
gyp ERR! node-gyp -v v3.0.3
gyp ERR! not ok

npm ERR! Windows_NT 10.0.10586
npm ERR! argv "C:\\Program Files (x86)\\nodejs\\node.exe" "C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "build-cpp-ffi"
npm ERR! node v5.5.0
npm ERR! npm  v3.3.12
npm ERR! code ELIFECYCLE
npm ERR! embed@1.0.0 build-cpp-ffi: `(cd src/native-cpp-ffi && node-gyp configure build)`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the embed@1.0.0 build-cpp-ffi script '(cd src/native-cpp-ffi && node-gyp configure build)'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the embed package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     (cd src/native-cpp-ffi && node-gyp configure build)
npm ERR! You can get their info via:
npm ERR!     npm owner ls embed
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\local_projects\rust\rust-in-node\npm-debug.log
```

Other native modules seems to work fine. It compiles C++ and rust nicely ceparately from each other, for some reason it can't reference the lib compiled from rust.

Please let me know if I can help you with it somehow.

The verbose output below:

```
npm run build-cpp-ffi --verbose
npm info it worked if it ends with ok
npm verb cli [ 'C:\\Program Files (x86)\\nodejs\\node.exe',
npm verb cli   'C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\bin\\npm-cli.js',
npm verb cli   'run',
npm verb cli   'build-cpp-ffi',
npm verb cli   '--verbose' ]
npm info using npm@3.3.12
npm info using node@v5.5.0
npm verb run-script [ 'prebuild-cpp-ffi', 'build-cpp-ffi', 'postbuild-cpp-ffi' ]
npm info lifecycle embed@1.0.0~prebuild-cpp-ffi: embed@1.0.0
npm info lifecycle embed@1.0.0~build-cpp-ffi: embed@1.0.0

> embed@1.0.0 build-cpp-ffi C:\local_projects\rust\rust-in-node
> (cd src/native-cpp-ffi && node-gyp configure build)


C:\local_projects\rust\rust-in-node\src\native-cpp-ffi>if not defined npm_config_node_gyp (node "C:\Program Files (x86)\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" configure build )  else (node  configure build )
gyp info it worked if it ends with ok
gyp verb cli [ 'C:\\Program Files (x86)\\nodejs\\node.exe',
gyp verb cli   'C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js',
gyp verb cli   'configure',
gyp verb cli   'build' ]
gyp info using node-gyp@3.0.3
gyp info using node@5.5.0 | win32 | ia32
gyp verb command configure []
gyp verb check python checking for Python executable "python2" in the PATH
gyp verb `which` failed Error: not found: python2
gyp verb `which` failed     at F (C:\Program Files (x86)\nodejs\node_modules\npm\node_modules\which\which.js:78:19)
gyp verb `which` failed     at E (C:\Program Files (x86)\nodejs\node_modules\npm\node_modules\which\which.js:82:29)
gyp verb `which` failed     at C:\Program Files (x86)\nodejs\node_modules\npm\node_modules\which\which.js:93:16
gyp verb `which` failed     at FSReqWrap.oncomplete (fs.js:82:15)
gyp verb `which` failed  python2 [Error: not found: python2]
gyp verb check python checking for Python executable "python" in the PATH
gyp verb `which` succeeded python C:\Python27\python.EXE
gyp verb check python version `python -c "import platform; print(platform.python_version());"` returned: "2.7.2\r\n"
gyp verb get node dir no --target version specified, falling back to host node version: 5.5.0
gyp verb command install [ '5.5.0' ]
gyp verb install input version string "5.5.0"
gyp verb install installing version: 5.5.0
gyp verb install --ensure was passed, so won't reinstall if already installed
gyp verb install version is already installed, need to check "installVersion"
gyp verb got "installVersion" 9
gyp verb needs "installVersion" 9
gyp verb install version is good
gyp verb get node dir target node version installed: 5.5.0
gyp verb build dir attempting to create "build" dir: C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\build
gyp verb build dir "build" dir needed to be created? null
gyp verb build/config.gypi creating config file
gyp verb build/config.gypi writing out config file: C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\build\config.gypi
gyp verb config.gypi checking for gypi file: C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\config.gypi
gyp verb common.gypi checking for gypi file: C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\common.gypi
gyp verb gyp gyp format was not specified; forcing "msvs"
gyp info spawn python
gyp info spawn args [ 'C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\gyp\\gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'msvs',
gyp info spawn args   '-G',
gyp info spawn args   'msvs_version=2015',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\local_projects\\rust\\rust-in-node\\src\\native-cpp-ffi\\build\\config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\Users\\andruhon\\.node-gyp\\5.5.0\\include\\node\\common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=C:\\Users\\andruhon\\.node-gyp\\5.5.0',
gyp info spawn args   '-Dnode_gyp_dir=C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\node_modules\\node-gyp',
gyp info spawn args   '-Dnode_lib_file=node.lib',
gyp info spawn args   '-Dmodule_root_dir=C:\\local_projects\\rust\\rust-in-node\\src\\native-cpp-ffi',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'C:\\local_projects\\rust\\rust-in-node\\src\\native-cpp-ffi\\build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp verb command build []
gyp verb build type Release
gyp verb architecture ia32
gyp verb node dev dir C:\Users\andruhon\.node-gyp\5.5.0
gyp verb found first Solution file build/binding.sln
gyp verb could not find "msbuild.exe" in PATH - finding location in registry
gyp verb "Release" dir needed to be created? null
gyp verb copying "node.lib" for ia32 C:\Users\andruhon\.node-gyp\5.5.0\Release\node.lib
gyp info spawn C:\Program Files (x86)\MSBuild\14.0\bin\msbuild.exe
gyp info spawn args [ 'build/binding.sln',
gyp info spawn args   '/nologo',
gyp info spawn args   '/p:Configuration=Release;Platform=Win32' ]
Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
Build started 27/01/2016 09:47:53.
Project "C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\build\binding.sln" on node 1 (default targets).
ValidateSolutionConfiguration:
  Building solution configuration "Release|Win32".
The target "BeforeGenerateProjectPriFile" listed in a BeforeTargets attribute at "C:\Program Files (x86)\MSBuild\Microsoft\NuGet\Microsoft.NuGet.targets (186,61)" does not exist in the project, and will be ignored.
Project "C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\build\binding.sln" (1) is building "C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\build\addon.vcxproj" (2) on node 1 (default targets).
InitializeBuildStatus:
  Touching "Release\obj\addon\addon.tlog\unsuccessfulbuild".
ClCompile:
  All outputs are up-to-date.
  All outputs are up-to-date.
Link:
  C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\bin\link.exe /ERRORREPORT:QUEUE /OUT:"C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\build\Release\addon.node" /INCREMENTAL:NO /NOLOGO kernel32.lib user32.lib gdi3
  2.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib DelayImp.lib "C:\Users\andruhon\.node-gyp\5.5.0\Release\node.lib" ..\..\..\rust\target\debug\embed.lib Delayimp.lib /DELAYLOAD
  :iojs.exe /DELAYLOAD:node.exe /MANIFEST /MANIFESTUAC:"level='asInvoker' uiAccess='false'" /manifest:embed /Debug /PDB:"C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\build\Release\addon.pdb" /MAP /MAPINFO:EXPORTS /OPT:
  REF /OPT:ICF /LTCG /TLBID:1 /DYNAMICBASE /NXCOMPAT /MACHINE:X86 /SAFESEH /ignore:4199 /DLL Release\obj\addon\addon.obj
  Release\obj\addon\win_delay_load_hook.obj
     Creating library C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\build\Release\addon.lib and object C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\build\Release\addon.exp
addon.obj : error LNK2001: unresolved external symbol "int __cdecl fibonacci(int)" (?fibonacci@@YAHH@Z) [C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\build\addon.vcxproj]
C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\build\Release\addon.node : fatal error LNK1120: 1 unresolved externals [C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\build\addon.vcxproj]
Done Building Project "C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\build\addon.vcxproj" (default targets) -- FAILED.

Done Building Project "C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\build\binding.sln" (default targets) -- FAILED.


Build FAILED.

"C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\build\binding.sln" (default target) (1) ->
"C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\build\addon.vcxproj" (default target) (2) ->
(Link target) ->
  addon.obj : error LNK2001: unresolved external symbol "int __cdecl fibonacci(int)" (?fibonacci@@YAHH@Z) [C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\build\addon.vcxproj]
  C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\build\Release\addon.node : fatal error LNK1120: 1 unresolved externals [C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\build\addon.vcxproj]

    0 Warning(s)
    2 Error(s)

Time Elapsed 00:00:00.73
gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\MSBuild\14.0\bin\msbuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Program Files (x86)\nodejs\node_modules\npm\node_modules\node-gyp\lib\build.js:270:23)
gyp ERR! stack     at emitTwo (events.js:100:13)
gyp ERR! stack     at ChildProcess.emit (events.js:185:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Windows_NT 10.0.10586
gyp ERR! command "C:\\Program Files (x86)\\nodejs\\node.exe" "C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "configure" "build"
gyp ERR! cwd C:\local_projects\rust\rust-in-node\src\native-cpp-ffi
gyp ERR! node -v v5.5.0
gyp ERR! node-gyp -v v3.0.3
gyp ERR! not ok

npm verb lifecycle embed@1.0.0~build-cpp-ffi: unsafe-perm in lifecycle true
npm verb lifecycle embed@1.0.0~build-cpp-ffi: PATH: C:\Program Files (x86)\nodejs\node_modules\npm\bin\node-gyp-bin;C:\local_projects\rust\rust-in-node\node_modules\.bin;C:\ProgramData\Oracle\Java\javapath;C:\Program Files (x86)\Intel\iCLS Client\;C:\Program Files\Intel\iCLS Client\;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\Program Files (x86)\Intel\OpenCL SDK\3.0\bin\x86;C:\Program Files (x86)\Intel\OpenCL SDK\3.0\bin\x64;C:\Program Files\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files\Intel\Intel(R) Management Engine Components\IPT;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\IPT;C:\Python27;C:\Python27\Scripts;C:\xampp\php;C:\Program Files (x86)\Git\cmd;C:\Program Files (x86)\Git\bin;C:\Program Files (x86)\Windows Live\Shared;C:\Program Files\Java\jdk1.7.0_67\bin;C:\Program Files\TortoiseSVN\bin;C:\adt-bundle-windows-x86_64-20140702\apache-ant-1.9.4\bin;C:\Program Files (x86)\nano-2.4.1-win32;C:\Go\bin;C:\Program Files (x86)\sbt\\bin;C:\Program Files (x86)\Skype\Phone\;C:\Program Files\Microsoft SQL Server\120\Tools\Binn\;C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit\;C:\Users\andruhon\AppData\Local\Android\sdk\tools;C:\Users\andruhon\AppData\Local\Android\sdk\platform-tools;C:\Program Files (x86)\nodejs\;C:\multirust\bin;C:\ProgramData\chocolatey\bin;c:\Users\andruhon\AppData\Local\atom\bin;C:\Users\andruhon\AppData\Local\atom\bin;C:\Go\bin;C:\Users\andruhon\AppData\Roaming\npm
npm verb lifecycle embed@1.0.0~build-cpp-ffi: CWD: C:\local_projects\rust\rust-in-node
npm info lifecycle embed@1.0.0~build-cpp-ffi: Failed to exec build-cpp-ffi script
npm verb stack Error: embed@1.0.0 build-cpp-ffi: `(cd src/native-cpp-ffi && node-gyp configure build)`
npm verb stack Exit status 1
npm verb stack     at EventEmitter.<anonymous> (C:\Program Files (x86)\nodejs\node_modules\npm\lib\utils\lifecycle.js:232:16)
npm verb stack     at emitTwo (events.js:100:13)
npm verb stack     at EventEmitter.emit (events.js:185:7)
npm verb stack     at ChildProcess.<anonymous> (C:\Program Files (x86)\nodejs\node_modules\npm\lib\utils\spawn.js:24:14)
npm verb stack     at emitTwo (events.js:100:13)
npm verb stack     at ChildProcess.emit (events.js:185:7)
npm verb stack     at maybeClose (internal/child_process.js:821:16)
npm verb stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
npm verb pkgid embed@1.0.0
npm verb cwd C:\local_projects\rust\rust-in-node
npm ERR! Windows_NT 10.0.10586
npm ERR! argv "C:\\Program Files (x86)\\nodejs\\node.exe" "C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "build-cpp-ffi" "--verbose"
npm ERR! node v5.5.0
npm ERR! npm  v3.3.12
npm ERR! code ELIFECYCLE
npm ERR! embed@1.0.0 build-cpp-ffi: `(cd src/native-cpp-ffi && node-gyp configure build)`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the embed@1.0.0 build-cpp-ffi script '(cd src/native-cpp-ffi && node-gyp configure build)'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the embed package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     (cd src/native-cpp-ffi && node-gyp configure build)
npm ERR! You can get their info via:
npm ERR!     npm owner ls embed
npm ERR! There is likely additional logging output above.
npm verb exit [ 1, true ]

npm ERR! Please include the following file with any support request:
npm ERR!     C:\local_projects\rust\rust-in-node\npm-debug.log
```

Still getting the same with VS 2013

```
npm run build-cpp-ffi --verbose
npm info it worked if it ends with ok
npm verb cli [ 'C:\\Program Files (x86)\\nodejs\\node.exe',
npm verb cli   'C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\bin\\npm-cli.js',
npm verb cli   'run',
npm verb cli   'build-cpp-ffi',
npm verb cli   '--verbose' ]
npm info using npm@3.3.12
npm info using node@v5.5.0
npm verb run-script [ 'prebuild-cpp-ffi', 'build-cpp-ffi', 'postbuild-cpp-ffi' ]
npm info lifecycle embed@1.0.0~prebuild-cpp-ffi: embed@1.0.0
npm info lifecycle embed@1.0.0~build-cpp-ffi: embed@1.0.0

> embed@1.0.0 build-cpp-ffi C:\local_projects\rust\rust-in-node
> (cd src/native-cpp-ffi && node-gyp configure build)


C:\local_projects\rust\rust-in-node\src\native-cpp-ffi>if not defined npm_config_node_gyp (node "C:\Program Files (x86)\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" configure build )  else (node  configure build )
gyp info it worked if it ends with ok
gyp verb cli [ 'C:\\Program Files (x86)\\nodejs\\node.exe',
gyp verb cli   'C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js',
gyp verb cli   'configure',
gyp verb cli   'build' ]
gyp info using node-gyp@3.0.3
gyp info using node@5.5.0 | win32 | ia32
gyp verb command configure []
gyp verb check python checking for Python executable "python2" in the PATH
gyp verb `which` failed Error: not found: python2
gyp verb `which` failed     at F (C:\Program Files (x86)\nodejs\node_modules\npm\node_modules\which\which.js:78:19)
gyp verb `which` failed     at E (C:\Program Files (x86)\nodejs\node_modules\npm\node_modules\which\which.js:82:29)
gyp verb `which` failed     at C:\Program Files (x86)\nodejs\node_modules\npm\node_modules\which\which.js:93:16
gyp verb `which` failed     at FSReqWrap.oncomplete (fs.js:82:15)
gyp verb `which` failed  python2 [Error: not found: python2]
gyp verb check python checking for Python executable "python" in the PATH
gyp verb `which` succeeded python C:\Python27\python.EXE
gyp verb check python version `python -c "import platform; print(platform.python_version());"` returned: "2.7.2\r\n"
gyp verb get node dir no --target version specified, falling back to host node version: 5.5.0
gyp verb command install [ '5.5.0' ]
gyp verb install input version string "5.5.0"
gyp verb install installing version: 5.5.0
gyp verb install --ensure was passed, so won't reinstall if already installed
gyp verb install version is already installed, need to check "installVersion"
gyp verb got "installVersion" 9
gyp verb needs "installVersion" 9
gyp verb install version is good
gyp verb get node dir target node version installed: 5.5.0
gyp verb build dir attempting to create "build" dir: C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\build
gyp verb build dir "build" dir needed to be created? null
gyp verb build/config.gypi creating config file
gyp verb build/config.gypi writing out config file: C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\build\config.gypi
gyp verb config.gypi checking for gypi file: C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\config.gypi
gyp verb common.gypi checking for gypi file: C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\common.gypi
gyp verb gyp gyp format was not specified; forcing "msvs"
gyp info spawn python
gyp info spawn args [ 'C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\gyp\\gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'msvs',
gyp info spawn args   '-G',
gyp info spawn args   'msvs_version=2013',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\local_projects\\rust\\rust-in-node\\src\\native-cpp-ffi\\build\\config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\Users\\andruhon\\.node-gyp\\5.5.0\\include\\node\\common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=C:\\Users\\andruhon\\.node-gyp\\5.5.0',
gyp info spawn args   '-Dnode_gyp_dir=C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\node_modules\\node-gyp',
gyp info spawn args   '-Dnode_lib_file=node.lib',
gyp info spawn args   '-Dmodule_root_dir=C:\\local_projects\\rust\\rust-in-node\\src\\native-cpp-ffi',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'C:\\local_projects\\rust\\rust-in-node\\src\\native-cpp-ffi\\build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp verb command build []
gyp verb build type Release
gyp verb architecture ia32
gyp verb node dev dir C:\Users\andruhon\.node-gyp\5.5.0
gyp verb found first Solution file build/binding.sln
gyp verb could not find "msbuild.exe" in PATH - finding location in registry
gyp verb "Release" dir needed to be created? null
gyp verb copying "node.lib" for ia32 C:\Users\andruhon\.node-gyp\5.5.0\Release\node.lib
gyp info spawn C:\Program Files (x86)\MSBuild\14.0\bin\msbuild.exe
gyp info spawn args [ 'build/binding.sln',
gyp info spawn args   '/nologo',
gyp info spawn args   '/p:Configuration=Release;Platform=Win32' ]
Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
Build started 27/01/2016 12:02:31.
Project "C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\build\binding.sln" on node 1 (
default targets).
ValidateSolutionConfiguration:
  Building solution configuration "Release|Win32".
The target "BeforeGenerateProjectPriFile" listed in a BeforeTargets attribute at "C:\Program F
iles (x86)\MSBuild\Microsoft\NuGet\Microsoft.NuGet.targets (186,61)" does not exist in the pro
ject, and will be ignored.
Project "C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\build\binding.sln" (1) is buil
ding "C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\build\addon.vcxproj" (2) on node
1 (default targets).
InitializeBuildStatus:
  Touching "Release\obj\addon\addon.tlog\unsuccessfulbuild".
ClCompile:
  All outputs are up-to-date.
  All outputs are up-to-date.
Link:
  C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\bin\link.exe /ERRORREPORT:QUEUE /OUT:
  "C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\build\Release\addon.node" /INCREMENT
  AL:NO /NOLOGO kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell
  32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib DelayImp.lib "C:\Users\andruhon\.node-gyp\
  5.5.0\Release\node.lib" ..\..\..\rust\target\release\embed.lib Delayimp.lib /DELAYLOAD:iojs.
  exe /DELAYLOAD:node.exe /MANIFEST /MANIFESTUAC:"level='asInvoker' uiAccess='false'" /manifes
  t:embed /DEBUG /PDB:"C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\build\Release\ad
  don.pdb" /MAP /MAPINFO:EXPORTS /OPT:REF /OPT:ICF /LTCG /TLBID:1 /DYNAMICBASE /NXCOMPAT /MACH
  INE:X86 /SAFESEH /ignore:4199 /DLL Release\obj\addon\addon.obj
  Release\obj\addon\win_delay_load_hook.obj
     Creating library C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\build\Release\add
  on.lib and object C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\build\Release\addon
  .exp
addon.obj : error LNK2001: unresolved external symbol "int __cdecl fibonacci(int)" (?fibonacci
@@YAHH@Z) [C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\build\addon.vcxproj]
C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\build\Release\addon.node : fatal error
LNK1120: 1 unresolved externals [C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\build\
addon.vcxproj]
Done Building Project "C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\build\addon.vcxp
roj" (default targets) -- FAILED.

Done Building Project "C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\build\binding.sl
n" (default targets) -- FAILED.


Build FAILED.

"C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\build\binding.sln" (default target) (1
) ->
"C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\build\addon.vcxproj" (default target)
(2) ->
(Link target) ->
  addon.obj : error LNK2001: unresolved external symbol "int __cdecl fibonacci(int)" (?fibonac
ci@@YAHH@Z) [C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\build\addon.vcxproj]
  C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\build\Release\addon.node : fatal erro
r LNK1120: 1 unresolved externals [C:\local_projects\rust\rust-in-node\src\native-cpp-ffi\buil
d\addon.vcxproj]

    0 Warning(s)
    2 Error(s)

Time Elapsed 00:00:00.68
gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\MSBuild\14.0\bin\msbuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Program Files (x86)\nodejs\node_modules\npm\node_modules\node-gyp\lib\build.js:270:23)
gyp ERR! stack     at emitTwo (events.js:100:13)
gyp ERR! stack     at ChildProcess.emit (events.js:185:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Windows_NT 10.0.10586
gyp ERR! command "C:\\Program Files (x86)\\nodejs\\node.exe" "C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "configure" "build"
gyp ERR! cwd C:\local_projects\rust\rust-in-node\src\native-cpp-ffi
gyp ERR! node -v v5.5.0
gyp ERR! node-gyp -v v3.0.3
gyp ERR! not ok

npm verb lifecycle embed@1.0.0~build-cpp-ffi: unsafe-perm in lifecycle true
npm verb lifecycle embed@1.0.0~build-cpp-ffi: PATH: C:\Program Files (x86)\nodejs\node_modules\npm\bin\node-gyp-bin;C:\local_projects\rust\rust-in-node\node_modules\.bin;C:\ProgramData\Oracle\Java\javapath;C:\Program Files (x86)\Intel\iCLS Client\;C:\Program Files\Intel\iCLS Client\;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\Program Files (x86)\Intel\OpenCL SDK\3.0\bin\x86;C:\Program Files (x86)\Intel\OpenCL SDK\3.0\bin\x64;C:\Program Files\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files\Intel\Intel(R) Management Engine Components\IPT;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\IPT;C:\Python27;C:\Python27\Scripts;C:\xampp\php;C:\Program Files (x86)\Git\cmd;C:\Program Files (x86)\Git\bin;C:\Program Files (x86)\Windows Live\Shared;C:\Program Files\Java\jdk1.7.0_67\bin;C:\Program Files\TortoiseSVN\bin;C:\adt-bundle-windows-x86_64-20140702\apache-ant-1.9.4\bin;C:\Program Files (x86)\nano-2.4.1-win32;C:\Go\bin;C:\Program Files (x86)\sbt\\bin;C:\Program Files (x86)\Skype\Phone\;C:\Program Files\Microsoft SQL Server\120\Tools\Binn\;C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit\;C:\Users\andruhon\AppData\Local\Android\sdk\tools;C:\Users\andruhon\AppData\Local\Android\sdk\platform-tools;C:\Program Files (x86)\nodejs\;C:\Program Files\Microsoft SQL Server\110\Tools\Binn\;C:\Program Files (x86)\Microsoft SDKs\TypeScript\1.0\;C:\multirust\bin;C:\ProgramData\chocolatey\bin;c:\Users\andruhon\AppData\Local\atom\bin;C:\Users\andruhon\AppData\Local\atom\bin;C:\Go\bin;C:\Users\andruhon\AppData\Roaming\npm
npm verb lifecycle embed@1.0.0~build-cpp-ffi: CWD: C:\local_projects\rust\rust-in-node
npm info lifecycle embed@1.0.0~build-cpp-ffi: Failed to exec build-cpp-ffi script
npm verb stack Error: embed@1.0.0 build-cpp-ffi: `(cd src/native-cpp-ffi && node-gyp configure build)`
npm verb stack Exit status 1
npm verb stack     at EventEmitter.<anonymous> (C:\Program Files (x86)\nodejs\node_modules\npm\lib\utils\lifecycle.js:232:16)
npm verb stack     at emitTwo (events.js:100:13)
npm verb stack     at EventEmitter.emit (events.js:185:7)
npm verb stack     at ChildProcess.<anonymous> (C:\Program Files (x86)\nodejs\node_modules\npm\lib\utils\spawn.js:24:14)
npm verb stack     at emitTwo (events.js:100:13)
npm verb stack     at ChildProcess.emit (events.js:185:7)
npm verb stack     at maybeClose (internal/child_process.js:821:16)
npm verb stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
npm verb pkgid embed@1.0.0
npm verb cwd C:\local_projects\rust\rust-in-node
npm ERR! Windows_NT 10.0.10586
npm ERR! argv "C:\\Program Files (x86)\\nodejs\\node.exe" "C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "build-cpp-ffi" "--verbose"
npm ERR! node v5.5.0
npm ERR! npm  v3.3.12
npm ERR! code ELIFECYCLE
npm ERR! embed@1.0.0 build-cpp-ffi: `(cd src/native-cpp-ffi && node-gyp configure build)`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the embed@1.0.0 build-cpp-ffi script '(cd src/native-cpp-ffi && node-gyp configure build)'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the embed package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     (cd src/native-cpp-ffi && node-gyp configure build)
npm ERR! You can get their info via:
npm ERR!     npm owner ls embed
npm ERR! There is likely additional logging output above.
npm verb exit [ 1, true ]

npm ERR! Please include the following file with any support request:
npm ERR!     C:\local_projects\rust\rust-in-node\npm-debug.log
```

--- a/README.md
+++ b/README.md
@@ -1,6 +1,6 @@
 # rust-in-node
 This is an comparison of different methods of calling `Rust` code from `Node` with benchmarks.
-You should have [`node@4.x.x`](https://nodejs.org/download/) and [`rust@1.1.0`](http://www.rust-lang.org/) installed and [`node-gyp`](https://github.com/TooTallNate/node-gyp/) configured.
+You should have [`node@4.x.x`](https://nodejs.org/download/) and [`rust@1.1.0`](http://www.rust-lang.org/)+ installed and [`node-gyp`](https://github.com/TooTallNate/node-gyp/) configured.
 
 ## Methods
 There are few different ways to call rust from node. All of them are besed on `FFI` ([Foreign Function Interface](https://doc.rust-lang.org/book/ffi.html))

--- a/README.md
+++ b/README.md
@@ -10,7 +10,7 @@ Minimal steps to crete dynamic library with rust
     cargo new embed
     cd embed
     edit src/lib.rs
-    
+
 ```rust
 #[no_mangle]
 pub extern fn fibonacci(n: i32) -> i32 {
@@ -21,7 +21,7 @@ pub extern fn fibonacci(n: i32) -> i32 {
 }
 ```
 
-    edit Cargo.toml 
+    edit Cargo.toml
 add to bottom
 ```toml
 [lib]
@@ -50,7 +50,7 @@ lib.fibonacci(10) // 89
 NOTE: path could be different
 
 ### Call dynamic library via c++ addon
-This is most complicate way because you have to write `c++` a bit. You can read more about native modules [`here`](https://nodejs.org/api/addons.html) 
+This is most complicate way because you have to write `c++` a bit. You can read more about native modules [`here`](https://nodejs.org/api/addons.html)
 
     mkdir cpp-ffi
     cd cpp-ffi
@@ -77,8 +77,8 @@ void init(Handle<Object> exports) {
 
 NODE_MODULE(addon, init)
 ```
-    
-    edit binding.gyp 
+
+    edit binding.gyp
 
 ```
 {
@@ -108,7 +108,7 @@ Then you can build module
     cd rust-in-node
     npm install
     npm run build
-    
+
 ## Benchmark
     node benchmark
 
@@ -121,6 +121,25 @@ nativeCppFFI.fibonacci(10)  x 1,825,865 ops/sec ±1.31% (89 runs sampled)
 ```
 
 As you can see the direct ffi call is to slow to have deal with it, but ffi + `C++` wrapper as fast as a native `C++` module, so `Rust` is good candidate for native modules for `Nodejs`
+
+## Building on windows
+Building on windows might be a challenging task, because `node-gyp` makes everyone
+unhappy on windows.
+
+If everything is configured properly `npm run build` should just work.
+
+However it is likely to be broken. Try these steps if it is:
+
+1. First ensure that you followed all windows installation instruction from README on https://github.com/nodejs/node-gyp
+2. Ensure that you using the same target for both Rust and C++. Rust should be
+compiled with MSVC target and target platform should be the same (ie i686/win32)
+3. Newer versions of [`Cargo`](https://github.com/rust-lang/cargo) produce `.dll.lib`
+files and older versions produce simply `.lib`. After building rust code please ensure
+that win embed lib name in `src\native-cpp-ffi\binding.gyp` match file names in `rust\target\release`.
+4. If you building only rust and native-cpp-ffi then you need to copy all
+libs compiled by cargo into directory of node addon. In this parcitular case:
+copy `rust/target/release/embed*` to `src/native-cpp-ffi/build/Release`
+
 
 ## Plans
 Test `Rust` module with multithreading. It could give even better result.

--- a/benchmark.js
+++ b/benchmark.js
@@ -1,6 +1,16 @@
 var Benchmark = require('benchmark');
 var suite = new Benchmark.Suite;
 var nativeRustFFI = require('./src/native-rust-ffi');
+/**
+ * In case if you need only C++ rust native call
+ * you need to copy all libs compiled by cargo
+ * into directory of node addon.
+ * In this parcitular case:
+ * copy rust/target/release/embed* to src/native-cpp-ffi/build/Release
+ *
+ * In this example it works without copying this files because ffi
+ * has included native-rust libs in the line of code above
+ */
 var nativeCppFFI = require('./src/native-cpp-ffi');
 var nativeCpp = require('./src/native-cpp');
 var vanilla = require('./src/vanilla');

--- a/src/native-cpp-ffi/addon.cc
+++ b/src/native-cpp-ffi/addon.cc
@@ -1,7 +1,7 @@
 #include <node.h>
+#include <addon.h>
 
 using namespace v8;
-extern int32_t fibonacci(int32_t input);
 
 void Method(const FunctionCallbackInfo<Value>& args) {
 	Isolate* isolate = Isolate::GetCurrent();

--- a/src/native-cpp-ffi/addon.h
+++ b/src/native-cpp-ffi/addon.h
@@ -1,0 +1,9 @@
+#ifdef WIN32
+extern "C" {
+#endif
+
+	extern int32_t fibonacci(int32_t input);
+
+#ifdef WIN32
+}
+#endif

--- a/src/native-cpp-ffi/binding.gyp
+++ b/src/native-cpp-ffi/binding.gyp
@@ -2,6 +2,9 @@
 	"targets": [{
 		"target_name": "addon",
 		"sources": ["addon.cc" ],
+        'include_dirs': [
+          '.',
+        ],
 		"conditions": [
 			['OS=="mac"', {
 				"libraries": [
@@ -10,7 +13,7 @@
 			}],
 			['OS=="win"', {
 				"libraries": [
-					"../../../rust/target/release/libembed.dll"
+					"../../../rust/target/release/embed.dll.lib"
 				]
 			}]
 		]

--- a/src/native-rust-ffi.js
+++ b/src/native-rust-ffi.js
@@ -1,5 +1,8 @@
 var ffi = require('ffi');
+var process = require('process')
 
-module.exports = ffi.Library('rust/target/release/libembed', {
+var isWin = /^win/.test(process.platform);
+
+module.exports = ffi.Library('rust/target/release/'+(!isWin?'lib':'')+'embed', {
 	fibonacci: ['int', ['int']]
 });

